R/functions-utils.R

```diff
@@ -604,16 +604,16 @@ rowRla <- function(x, group, log.transform = TRUE) {
 #' @md
 #'
 #' @noRd
-.update_feature_definitions <- function(x, original_names, subset_names,
-                                         BPPARAM = bpparam()) {
+.update_feature_definitions <- function(x, original_names, subset_names) {
     ## Skip if they are the same.
     if (length(original_names) == length(subset_names) &&
         all.equal(original_names, subset_names))
         return(x)
-    x$peakidx <- bplapply(x$peakidx, function(z) {
-        idx <- base::match(original_names[z], subset_names)
-        idx[!is.na(idx)]
-    }, BPPARAM = BPPARAM)
+    f <- as.factor(rep(seq_len(length(x$peakidx)), lengths(x$peakidx)))
```

Comment:
Rationale here was that `bpapply` had too much overhead for that simple task ?

Reply:
`bplapply` (and also `lapply`) was just very slow, not really because of the parallel processing overlap but more because a `match` call against all names of chromatographic peaks was performed for **every** feature. This version is now vectorized performing the `match` in one go. Calling e.g. `filterFile` with `keepFeatures = TRUE` on a data set with ~ 5000 features across 10 files took ~ 30 seconds with the old version and now it takes less than a second. The performance gain is expected to increase with data set size.